pkg/ops/transferOps.go

```diff
@@ -75,16 +75,12 @@ func Approve(ctx context.Context, logger log.Logger, _spender common.Address, am
 	return nil
 }
 
-func Balance(ctx context.Context, addr common.Address) error {
-	client := ctx.Value(tellorCommon.ClientContextKey).(rpc.ETHClient)
-
+func Balance(ctx context.Context, client rpc.ETHClient, getterInstance *getter.TellorGetters, addr common.Address) error {
 	ethBalance, err := client.BalanceAt(context.Background(), addr, nil)
```

Comment:
```suggestion
	ethBalance, err := client.BalanceAt(ctx, addr, nil)
```

The idea is that here the context will be created in the main package and then passed to the this component, this way the main package can cancel the context and this will propagate here and will cancel any pending requests and the function will return. Withoug this there is no way for the main package to control when the child function/components need to exit.

Here `context.Background()` creates a new empty context, but instead we should just use the one provided in the `Balance(ctx context.Context, `